### PR TITLE
Fix #76,#71

### LINKED ISSUE #76: How can i update the generated schema.json?
Hey im currently playing around with houdini, trying mutations etc. and learning stuff but it seems like the schema.json doesn't get updated when i run generate? Does houdini currently offer a way to update the schema? Or do i have to update it myself?

I could probably create a pull request in the next days, lets say we pass `--pull` to the generate command to pull the newest schema before generating?

### PR DESCRIPTION
**Pull request by pixelmund**
Hey, this is my first take on implementing the #76 `pull schema` functionality.
I also included support to read from .graphql and .gql #71 as schemaPath, because i was already on that topic.

Im open for any suggestions and changes, i don't really know how to write tests for this or if you have any tests for commands currently.

### CODE REVIEW

File: packages/houdini/cmd/main.ts [hunk 1]
```diff
@@ -14,11 +16,30 @@ const program = new Command()
 program
 	.command('generate')
 	.description('generate the application runtime')
-	.action(async () => {
+	.option('-p|--pull-schema', 'Pull the latest schema before generating')
```
Comment (AlecAivazis) [inline packages/houdini/cmd/main.ts:19]: Let's change this to `-p, --pull-schema` in order to match the `help` command: 
image

File: packages/houdini/cmd/utils/getSchema.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		// write the schema file
+		await fs.writeFile(
+			schemaPath,
+			JSON.stringify(JSON.parse(content).data),
+			'utf-8'
+		)
+	} catch (e) {
+		console.log('encountered error parsing response as json: ' + e.message)
+		console.log('full body: ' + content);
+        process.exit(0);
```
Comment (AlecAivazis) [inline packages/houdini/cmd/utils/getSchema.ts:26]: Looks like prettier needs to be run

File: packages/houdini/cmd/utils/getSchema.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { getIntrospectionQuery } from 'graphql'
+import fs from 'fs/promises'
+import fetch from 'node-fetch'
+
+export async function getSchema(url: string, schemaPath: string) {
```
Comment (AlecAivazis) [inline packages/houdini/cmd/utils/getSchema.ts:5]: Thought on renaming this to `writeSchema`? `get` makes me think that the function will return the value

**Review by AlecAivazis — CHANGES_REQUESTED** [review]

**Comment by AlecAivazis** [comment]
Thanks for opening this up @pixelmund - just a few changes but we should be able to get this in quickly

File: packages/houdini/cmd/utils/getSchema.ts (hunk 3 above)
Comment (pixelmund) [reply packages/houdini/cmd/utils/getSchema.ts]: You're right i changed it to `writeSchema`.

File: example/houdini.config.js [hunk 4]
```diff
@@ -2,6 +2,9 @@ import path from 'path'
 
 export default {
 	schemaPath: path.resolve('./schema/introspection.json'),
```
Comment (AlecAivazis) [inline example/houdini.config.js:4]: It looks like you removed the `.json` file here but left it in the config. Should this point to `.schema/schema.gql`?

Comment (pixelmund) [reply example/houdini.config.js:4]: The introspection.json file is still in the schema folder isn't it? I can remove one of them if you like, it was just for demonstration

Comment (AlecAivazis) [reply example/houdini.config.js:4]: Ahh i see what happened - got confused by github's diff. There is a single green line at the bottom of a lot of red haha. Ignore this :)

**Review by AlecAivazis — APPROVED** [review]
Looks good! Thanks for getting this changes out so quickly

**Comment by AlecAivazis** [comment]
Just deployed this as `0.5.1` and took the opportunity to remove the `@kit` version tag.

Thanks again!

**Comment by pixelmund** [comment]
Happy to help!

**Pull request merged**